src/jit/codegenlegacy.cpp

```diff
@@ -19462,14 +19462,30 @@ regMaskTP CodeGen::genCodeForCall(GenTreePtr call, bool valUsed)
                         regNumber indCallReg;
 
                         case IAT_VALUE:
+                        {
                             //------------------------------------------------------
                             // Non-virtual direct calls to known addressess
                             //
                             // The vast majority of calls end up here....  Wouldn't
                             // it be nice if they all did!
                             CLANG_FORMAT_COMMENT_ANCHOR;
 #ifdef _TARGET_ARM_
-                            if (!arm_Valid_Imm_For_BL((ssize_t)addr))
+                            // We may use direct call for some of recursive calls
+                            // as we can safely estimate the distance from the call site to the top of the method
+                            const int MAX_PROLOG_SIZE = MAX_USHORT_SIZE_T;
+                            const int codeOffset      = getEmitter()->emitCurCodeOffset + getEmitter()->emitCurIGsize;
+                            if (compiler->gtIsRecursiveCall(call->AsCall()) &&
+                                MAX_PROLOG_SIZE + codeOffset + 4 < 16777216) // if code offset fits in imm24
+                            {
+                                getEmitter()->emitIns_Call(emitter::EC_FUNC_TOKEN, methHnd,
+                                                           INDEBUG_LDISASM_COMMA(sigInfo) NULL, args, retSize,
+                                                           gcInfo.gcVarPtrSetCur, gcInfo.gcRegGCrefSetCur,
+                                                           gcInfo.gcRegByrefSetCur, ilOffset, REG_NA, REG_NA, 0,
+                                                           0,     /* ireg, xreg, xmul, disp */
+                                                           false, /* isJump */
```

Comment:
Like Carol said, you may need to split the lines another way so that the formatter doesn't complain. It's basically just saying that the lines are too long, and you need to split the lines (max line length is 120). You can also look at the instructions on https://github.com/dotnet/jitutils/blob/master/doc/getstarted.md#formatting-jit-source to see how to use the formatter in fix mode.